framework/include/utils/InputParameters.h

```diff
@@ -849,7 +849,7 @@ InputParameters::rangeCheck(const std::string & full_name,
         iss.seekg(short_name.size() + 1);
 
         size_t index;
-        if (iss >> index)
+        if (iss >> index && iss.eof())
```

Comment:
What are the `eof()` checks here doing? Making sure that there's only a single number in the stream with nothing after it?

Reply:
We discussed this, the following code sets `iss.eof()`, which is what we are checking for. 
```
std::istringstream iss("42");
int x;
iss >> x;
```